pkgs/build-support/fetchnextcloudapp/default.nix

```
@@ -0,0 +1,39 @@
+{ stdenv, gnutar, findutils, fetchurl, ... }:
+{ name
+, url
+, version
+, sha256
+, patches ? [ ]
+}:
+stdenv.mkDerivation {
+  name = "nc-app-${name}";
+  inherit version patches;
+
+  src = fetchurl {
+    url = url;
+    sha256 = sha256;
+  };
+
+  nativeBuildInputs = [
+    gnutar
+    findutils
+  ];
+
+  unpackPhase = ''
+    tar -xzpf $src
+    find -type f -executable -exec chmod o-x,g-x-w,a-x-w '{}' \;
+  '';
+
+  installPhase = ''
+    approot="$(dirname $(dirname $(find -path '*/appinfo/info.xml' | head -n 1)))"
+
+    if [ -d "$approot" ];
+    then
+      mv "$approot/" $out
+      chmod -R g-w $out
```

This shouldn't be needed. `/nix/store` is read-only anyways.

---

I added this because nix was complaining about "suspicious permissions" (or something like that)

---

I've been experimenting with this a bit, and it seems that I even need:
```suggestion
      chmod -R a-w $out
```

For the tasks app (since its most recent version)

---

It seems that this is done by Nix to avoid having other processes writing into a store-path that's about to be created and with `cp` the permissions would be different. As suggested above, the `chmod` should be sufficient to fix this entirely.

---

Done in 98f95f9